front/src/modules/pathfinding/components/IncompatibleConstraints/IncompatibleConstrainstFilters.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useCallback, type FC, type HTMLAttributes } from 'react';
+
+import { Filter } from '@osrd-project/ui-icons';
+import cx from 'classnames';
+import { toPairs } from 'lodash';
+import { useTranslation } from 'react-i18next';
+
+import type { IncompatibleConstraintTypes } from './type';
+import { getIcon } from './utils';
+
+interface IncompatibleConstraintsFiltersProps extends HTMLAttributes<unknown> {
+  data: Record<IncompatibleConstraintTypes, { count: number; enabled: boolean }>;
+  toggle: (type: string) => void;
+}
+const IncompatibleConstraintsFilters: FC<IncompatibleConstraintsFiltersProps> = (props) => {
+  const { t } = useTranslation(['operationalStudies/manageTrainSchedule']);
+  const { data, toggle, ...htmlAttrs } = props;
+  const allSelected = toPairs(data).every(([, e]) => e.enabled);
+
+  const getTitle = useCallback(
+    (label: string, enabled: boolean) => {
+      if (allSelected) {
+        return t('incompatibleConstraints.filters.focus', { name: label });
+      }
+      return enabled
+        ? t('incompatibleConstraints.filters.disable', { name: label })
+        : t('incompatibleConstraints.filters.enable', { name: label });
+    },
+    [t, allSelected]
```

Comment:
Maybe you can merge these. And here, we could also have check when only one filter is enable ?

```suggestion
  const allSelected = toPairs(data).every(([, e]) => e.enabled);

  const getTitle = useCallback(
    (label: string, enabled: boolean) => {
    const allSelected = Object.values(data).every((value) => value.enabled);
      if (allSelected) {
        return t('incompatibleConstraints.filters.focus', { name: label });
      }
      
      if (oneEnabled) return t('incompatibleConstraints.filters.activateAll'
      
      return enabled
        ? t('incompatibleConstraints.filters.disable', { name: label })
        : t('incompatibleConstraints.filters.enable', { name: label });
    },
    [t, data]
```